### PR TITLE
Add the Test Plan doc for PFC Storm with Shared Headroom 

### PR DESCRIPTION
**Pull request by vivekrnv**

### CODE REVIEW

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# PFC Storm with Shared Headroom Test Plan
+
+## Motivation
+
+When a PFC Storm is seen, PFC WatchDog on the Melanox Devices applies Zero Buffer Profiles on Ingress PG and Egress Queues. However if the buffer occupany has already crossed Shared Headroom (i.e. Shared Buffer + Xon) when a PFC Storm is detected, a bug was recently found on mellanox platform.
```
Comment (stephenxs) [inline doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md]: Not sure whether or not necessary to say `a bug was found on Mellanox platform` ...

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
++ This test case requires an RPC image
+
+## Test Plan
++ Verify if the shared headroom is enabled
++ Make sure buffer occupancy crosses into the shared headroom region
+   - Achieve buffer congestion by closing the dut tx port using `sai_thrift_port_tx_disable` API: https://github.com/Azure/sonic-mgmt/blob/master/tests/saitests/switch.py#L624.
+   - Send pkts from the PTF docker which are destined to egress out of the dut tx port.
+   - Make sure to send atleast num_pkts_pfs_frame + private_headroom pkts pkts
+   - num_pkts_pfs_frame: num of pkts required to be sent in order to trigger a PFC frame from the DUT. More on this here: https://github.com/Azure/sonic-mgmt/blob/master/tests/qos/files/mellanox/qos_param_generator.py
+   - private_headroom_pkts is specific to mellanox which is in the order of a few pkts.
```
Comment (stephenxs) [inline doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:21]: maybe we need to add this to `qos_param_generator.py` and the test parameters passed from `tests/qos/test_qos_sai.py` 
(which runs from sonic-mgmt docker) to `saitest/sai_qos_test.py` (which runs from ptf docker).

Comment (stephenxs) [inline doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:21]: better to check the counters as well.

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    
++ Trigger a PFC storm directed towards the DUT Rx port
++ PFC Watchdog is triggered
```
Comment (stephenxs) [inline doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:25]: Do we need to 
- increase/decrease the counters during PFC watchdog triggered?
- check whether the PFC frames stopped exactly once the buffer occupancy drops below the xon threshold?

To achieve that we need more than one egress ports.

**Comment by stephenxs** [comment]
Another idea is to leverage `PFCXONtest`, just adding an extra argument indicating whether PFC storm should be started on the ingress port. By doing so, we can implement the checks more easily

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md (hunk 3 above)
Comment (vivekkarri) [reply doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:25]: The Xon test already covers this case right, the purpose of this case is only to check if the PFC frames are not being sent after the threshold drops

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md (hunk 1 above)
Comment (vivekkarri) [reply doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md]: Will remove

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md (hunk 2 above)
Comment (vivekkarri) [reply doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:21]: Yes, I do. Is the value static/fixed per platform or should it be calculated dynamically?

**Comment by vivekrnv** [comment]
> PFC

Hmm that seems like a good idea.

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md (hunk 2 above)
Comment (vivekkarri) [reply doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:21]: Added

File: doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md (hunk 3 above)
Comment (vivekkarri) [reply doc/pfc_watchdog/Pfc_Storm_with_Shared_Headroom_test_plan.md:25]: So, just using one port

**Pull request closed without merging**